docroot/modules/custom/uiowa_core/uiowa_core.module

```diff
@@ -134,6 +135,141 @@ function template_preprocess_uiowa_core_headline(&$variables) {
 
 }
 
+/**
+ * Implements hook_preprocess_media().
+ */
+function uiowa_core_preprocess_media(&$variables) {
+
+  // Detect if this is a figure and if it is,
+  // it has already been processed for inline sizing and alignment.
+  $is_figure = FALSE;
+
+  if (isset($variables['attributes']['data-figure'])) {
+    $is_figure = filter_var($variables['attributes']['data-figure'], FILTER_VALIDATE_BOOLEAN);
+  }
+
+  // Check to see if the media element has been processed, or if it is an embed.
+  if (!$is_figure && isset($variables['elements']['#embed']) && $variables['elements']['#embed']) {
```

Comment:
Do we think there'll ever be a situation where we have a figure that was processed, but might still need further processing in this?

If not, can we shorten this to just exit if it's a figure, and if we hit the check further down, we already know it's not a figure.
```suggestion
/**
 * Implements hook_preprocess_media().
 */
function uiowa_core_preprocess_media(&$variables) {

  // Detect if this is a figure and if it is,
  // it has already been processed for inline sizing and alignment.
  if (isset($variables['attributes']['data-figure']) &&
    filter_var($variables['attributes']['data-figure'], FILTER_VALIDATE_BOOLEAN) === FALSE) {
    return;
  }

  // Check to see if the media element has been processed, or if it is an embed.
  if (!$is_figure && isset($variables['elements']['#embed']) && $variables['elements']['#embed']) {
```

Reply:
I agree with this change